rasa/shared/core/domain.py

```diff
@@ -1215,3 +1221,81 @@ def slot_mapping_for_form(self, form_name: Text) -> Dict[Text, Any]:
             The slot mapping or an empty dictionary in case no mapping was found.
         """
         return self.forms.get(form_name, {})
+
+    @staticmethod
+    def _find_events_after_actions(
```

Comment:
Naah. I this has nothing to do with the `domain` in my opinion. Let's keep this in the `tracker` module. Just for the sake of keeping imports short we shouldn't move things to the wrong places.

Reply:
actually, it doesn't do anything with trackers either (that's why I moved it)

Reply:
I think it should go here: https://github.com/RasaHQ/rasa/blob/master/rasa/core/training/__init__.py

Reply:
I think having any code aside from imports in the `__init__.py` file is bad practice.

> actually, it doesn't do anything with trackers either (that's why I moved it)

it receives a list of trackers and goes through the conversation events of the trackers. I think it's very closely tied to trackers and their content.

Reply:
yes, but `trackers.py` deals with individual tracker. List of trackers is training data

Reply:
some resources on __init__.py and good practices https://docs.python-guide.org/writing/structure/#packages

Reply:
In my opinion it is much closer to domain than to trackers, because it creates a signature for actions which is basically part of domain

Reply:
there were other methods in `init.py`, totally up for moving all of them out

Reply:
funny enough, many of them are not used 😅

Reply:
> yes, but trackers.py deals with individual tracker. List of trackers is training data

But the resulting fingerprints are also used during prediction time (in `processor`). 

> In my opinion it is much closer to domain than to trackers, because it creates a signature for actions which is basically part of domain

But they only use the slot info from the domain. All other info is events and trackers. The only other place I can imagine is the module of `TrackerWithCachedStates` 

> there were other methods in init.py, totally up for moving all of them out

🎉  Let's do that in a different PR though. If the stuff isn't used -> feel free to delete it

Reply:
> But they only use the slot info from the domain. All other info is events and trackers. The only other place I can imagine is the module of TrackerWithCachedStates

the other place would be `policy.py` that's actually who deals with events list of trackers and domain

`trackers.py` don't deal with a list of trackers

Reply:
> But the resulting fingerprints are also used during prediction time
but not the method itself

Reply:
thinking about it more. I'd keep it in training, that's the one that deal with training data represented as list of trackers and that the main thing about this function